File: backend/go/internal/domain/car.go
package domain

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Car struct {
	ID        uuid.UUID      `json:"id" gorm:"type:uuid;primaryKey" example:"550e8400-e29b-41d4-a716-446655440000"`
	Brand     string         `json:"brand" gorm:"not null;size:100" example:"Toyota"`
	Model     string         `json:"model" gorm:"not null;size:100" example:"Corolla"`
	Year      int            `json:"year" gorm:"not null" example:"2024"`
	Color     string         `json:"color" gorm:"not null;size:50" example:"White"`
	Price     float64        `json:"price" gorm:"not null" example:"35000.00"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `json:"-" gorm:"index"`
}

type CreateCarRequest struct {
	Brand string  `json:"brand" example:"Toyota"`
	Model string  `json:"model" example:"Corolla"`
	Year  int     `json:"year" example:"2024"`
	Color string  `json:"color" example:"White"`
	Price float64 `json:"price" example:"35000.00"`
}

type UpdateCarRequest struct {
	Brand *string  `json:"brand,omitempty" example:"Honda"`
	Model *string  `json:"model,omitempty" example:"Civic"`
	Year  *int     `json:"year,omitempty" example:"2025"`
	Color *string  `json:"color,omitempty" example:"Black"`
	Price *float64 `json:"price,omitempty" example:"40000.00"`
}

func (c *Car) BeforeCreate(tx *gorm.DB) error {
	if c.ID == uuid.Nil {
		c.ID = uuid.New()
	}
	return nil
}

File: frontend/react/src/types/car.ts
export interface Car {
    id: string;
    brand: string;
    model: string;
    year: number;
    color: string;
    price: number;
    created_at: string;
    updated_at: string;
}

export interface CreateCarRequest {
    brand: string;
    model: string;
    year: number;
    color: string;
    price: number;
}

export interface UpdateCarRequest {
    brand?: string;
    model?: string;
    year?: number;
    color?: string;
    price?: number;
}

export interface SuccessResponse<T> {
    data: T;
}

export interface PaginatedResponse<T> {
    data: T[];
    total: number;
    offset: number;
    limit: number;
}

export interface ErrorResponse {
    error: string;
}

export interface ValidateRequest {
    api_key: string;
}

export interface ValidateResponse {
    token: string;
}
